doc/spec/tags.md

```diff
@@ -0,0 +1,171 @@
+# Tags and Signatures
+
+## Introduction
+
+Classically, the role of tagging has been less formal in the docker eco-
+system. Effectively, the concept of "tagging" meant that one assigned a name
+to an image id.
+
+Tags in docker are missing several features that one might expect when coming
+from other systems. In git, one can tag any commit and then sign that tag,
+providing a name to a given revision and the ability to verify that the name
+was assigned by a trusted party. We'd like bring this more familar model to
+the docker ecosystem.
+
+The following changes are proposed:
+
+1. Tags will become first-class object in the image distribution data model.
+2. Tags are the primary mechanism for verification and named provenance.
+
+The tag object's role will be equivalent to the v2, schema version 1.
+
+## Tags
+
+A tag is simply a named pointer to content. The content can be any blob but
+should mostly be a manifest. One can sign tags to later verify that they were
+created by a trusted party.
+
+This also relies on signatures being a first-class component. We may explore
+that in a separate proposal or integrate signatures completely with tags.
+
+### Routes
+
+The following routes are added to the V2 API specification:
+
+|Method|Path|Entity|Description|
+-------|----|------|------------
+| GET    | `/v2/<name>/tags/`         | Tag | List the tags for the repo. Must end with slash.      |
+| GET    | `/v2/<name>/tags/<tag>`    | Tag | Return the tag object identified by `name` and `tag`. |
+| PUT    | `/v2/<name>/tags/<tag>`    | Tag | Put the tag identified by `name` and `tag`.           |
+| DELETE | `/v2/<name>/tags/<tag>`    | Tag | Delete the tag identified by `name` and `tag`.        |
+
+Note that for compatibility with the first version 2 specification, requests
+to "/v2/<name>/tags/<tag>" will require an accept header with the value of
+"application/vnd.docker.distribution.tag.v1+json" to disambiguate from the
+endpoint "/v2/<name>/tags/list".
+
+### Media Types
+
+Tag's may have the following media types:
+
+| Media Type                                             | Description                                      |
+---------------------------------------------------------|--------------------------------------------------|
+| `application/vnd.docker.distribution.tag.v1+json`      | Base tag object                                  |
+| `application/vnd.docker.distribution.tag.v1+jws`       | tag object wrapped in one or more jws signatures |
+| `application/vnd.docker.distribution.tag.v1+prettyjws` | tag object with jws signatures in pretty format  |
```

Comment:
prettyjws and jws are formats that differ in more than just whitespace. One has the payload embedded as json while the other has it encoded in base64.